source/reference/authentication.txt

```diff
@@ -0,0 +1,116 @@
+.. _c2c-authentication:
+
+=================================================
+Authentication Using Workload Identity Federation
+=================================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: twocols
+
+.. include:: /includes/mongosync-and-oidc.rst
+
+The source and destination clusters must use MongoDB 7.0 or later.
+
+You can use ``mongosync`` to sync data between source and destination
+clusters and authenticate the cluster connections using various
+combinations:
+
+- Sync source and destination clusters both authenticated using Workload
+  Identity Federation.
+- Sync a source cluster authenticated using Workload Identity Federation
+  and a destination cluster that doesn't use Workload Identity
+  Federation.
+- Sync a destination cluster authenticated using Workload Identity
+  Federation and a source cluster that doesn't use Workload Identity
+  Federation.
+
+``mongosync`` supports replica sets and sharded clusters.
+
+Examples
+--------
+
+In the :ref:`connection string <connections-connection-options>`, set
+:urioption:`authMechanism` to ``MONGODB-OIDC`` and set
+:urioption:`authMechanismProperties` as needed:
+
+- For Microsoft Azure, set ``authMechanismProperties`` to
+  ``ENVIRONMENT:azure``
+- For Google Cloud Platform, set ``authMechanismProperties`` to
+  ``ENVIRONMENT:gcp``
+
+For more details about:
+
+- OIDC in connections strings, see
+  :ref:`connection-string-auth-options`
+- Workforce Identity Federation and OIDC in Atlas, see
+  :atlas:`Set up Workforce Identity Federation with OIDC
+  </workforce-oidc/#std-label-oidc-authentication-workforce/>`
+
+Connect to MongoDB Clusters Using Microsoft Azure Instance Metadata Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following ``mongosync`` example connects to MongoDB clusters using
+Microsoft Azure Instance Metadata Service (IMDS):
+
+.. code-block:: shell
+
+   ./bin/mongosync \
+         --logPath /var/log/mongosync \
+         --cluster0 "mongodb://clusterAdmin:superSecret@clusterOne01.fancyCorp.com:20020,clusterOne02.fancyCorp.com:20020,clusterOne03.fancyCorp.com:20020/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure" \
+         --cluster1 "mongodb://clusterAdmin:superSecret@clusterTwo01.fancyCorp.com:20020,clusterTwo02.fancyCorp.com:20020,clusterTwo03.fancyCorp.com:20020/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure"
```

Comment:
Should there be a username/password in the connection string when authenticating via OIDC? (cc @pmeredit)